utils/checks.stanza

```diff
@@ -113,8 +133,207 @@ public pcb-check rated-temperature (component:JITXObject):
     locators =             [component]  
   )
 
-public pcb-check io (module:JITXObject) :
+; Returns all the pins that are electrically-meaningful, ie are not 
+; from components marked do-not-place
+defn all-pins (module:JITXObject) -> Tuple<JITXObject> : 
+  populated-components(module) 
+    $> seq-cat{pins, _}
+    $> to-tuple
+  
+; Checks if a pin is marked "no-connect"
+defn no-connect? (p:JITXObject) -> True|False : 
+  has-property?(p.no-connect) or
+  has-property?(p.nc) or 
+  has-property?(p.NC)
+
+; Returns if a pin is a digital-io pin
+defn is-digital-io-pin? (p:JITXObject) -> True|False : 
```

Comment:
We probably need to split this into a `is-digital-output-pin?` and `is-digital-io-pin?` as that would facilitate the checks for avoiding multiple outputs on the same net but allowing multiple io pins.

Reply:
Updated!